### PR TITLE
Coverage Patch - trying to simulate native events

### PR DESCRIPTION
**Pull request by felquis**

### CODE REVIEW

File: src/EveryExample.js [hunk 1]
```diff
@@ -30,8 +30,8 @@ export default class EveryExample extends Component {
       <div>
         <p style={{color: this.state.color}}>Total time: {this.state.displayTime}</p>
 
-        <Every frame do={(dt) => this.handleTotalTime(dt)} />
-        <Every s="1" ms="500" do={() => this.handleColor()} />
+        <Every frame do={this.handleTotalTime.bind(this)} />
+        <Every s="1" ms="500" do={this.handleColor.bind(this)} />
```
Comment (felquis) [inline src/EveryExample.js]: Not sure why I did it

File: src/TouchExample.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ export default class TouchExample extends Component {
     })
   }
 
-  check = (event) => {
+  check(event) {
```
Comment (felquis) [inline src/TouchExample.js]: It was breaking the tests

File: src/spec/DeviceOrientationExample.test.js [hunk 3]
```diff
@@ -1,8 +1,20 @@
...
+  render(<DeviceOrientationExample />, div)
 })
+
+it('calls handleDeviceOrientation', () => {
+  sinon.spy(DeviceOrientationExample.prototype, 'handleDeviceOrientation')
+  const wrapper = mount(<DeviceOrientationExample />)
+  const e = document.createEvent('CustomEvent')
+  e.initEvent('deviceorientation', true, true)
+  window.dispatchEvent(e)
+  expect(DeviceOrientationExample.prototype.handleDeviceOrientation.calledOnce).toBe(true)
```
Comment (felquis) [inline src/spec/DeviceOrientationExample.test.js]: In this crazy manner I could simulate deviceorientation event, but I couldn't make the same for touch events or geolocation :/

File: src/spec/GeolocationExample.test.js [hunk 4]
```diff
@@ -1,8 +1,34 @@
 import React from 'react'
-import ReactDOM from 'react-dom'
+import { render } from 'react-dom'
+import { shallow } from 'enzyme'
 import GeolocationExample from '../GeolocationExample'
+import sinon from 'sinon'
+import geolocate from 'mock-geolocation'
```
Comment (felquis) [inline src/spec/GeolocationExample.test.js]: It does not work :(

File: src/spec/GeolocationExample.test.js [hunk 5]
```diff
@@ -1,8 +1,34 @@
...
+  geolocate.restore()
+})
+
+// it('calls componentDidMount', () => {
+//   sinon.spy(GeolocationExample.prototype, 'handleLocationChange')
+//   const wrapper = shallow(<GeolocationExample />)
+//   global.geolocation.watchPosition()
+
+//   expect(GeolocationExample.prototype.handleLocationChange.calledOnce).toBe(true)
+// })
```
Comment (felquis) [inline src/spec/GeolocationExample.test.js]: Sorry :wastebasket:

File: src/spec/TouchExample.test.js [hunk 6]
```diff
@@ -1,8 +1,18 @@
...
   const div = document.createElement('div')
-  ReactDOM.render(<TouchExample />, div)
+  render(<TouchExample />, div)
 })
+
+it('calls handleTouch', () => {
+  sinon.spy(TouchExample.prototype, 'handleTouch')
+  const wrapper = mount(<TouchExample />)
+  wrapper.simulate('touchStart')
+  expect(TouchExample.prototype.handleTouch.calledOnce).toBe(true)
```
Comment (felquis) [inline src/spec/TouchExample.test.js]: Failed

File: src/spec/WindowResizeExample.test.js [hunk 7]
```diff
@@ -1,8 +1,17 @@
...
 it('renders without crashing', () => {
   const div = document.createElement('div')
-  ReactDOM.render(<WindowResizeExample />, div)
+  render(<WindowResizeExample />, div)
+})
+
+it('calls componentDidMount', () => {
+  sinon.spy(WindowResizeExample.prototype, 'handleWindowResize')
+  const wrapper = mount(<WindowResizeExample />)
+  window.dispatchEvent(new Event('resize'))
```
Comment (felquis) [inline src/spec/WindowResizeExample.test.js]: failed again!

File: src/EveryExample.js (hunk 1 above)
Comment (bukinoshita) [inline src/EveryExample.js]: By binding your event handlers like this, you create brand new functions every time React calls your component’s render

Comment (felquis) [inline src/EveryExample.js]: OMG, I didn't know that! Thanks! :D

**Comment by felquis** [comment]
Closing it since we have long moved on.

**Pull request closed without merging**